File: spring-web-java/src/main/java/spring/web/java/presentation/exception/DuplicateRequestException.java
package spring.web.java.presentation.exception;

import org.springframework.http.HttpStatus;

public class DuplicateRequestException extends AbstractHttpException {

	public DuplicateRequestException(Long memberId, String method, String uri) {
		super("회원(ID: %d)의 요청이 중복되었습니다. (%s %s)".formatted(memberId, method, uri), HttpStatus.TOO_MANY_REQUESTS);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/exception/DuplicateRequestException.kt
package spring.web.kotlin.presentation.exception

import org.springframework.http.HttpStatus

class DuplicateRequestException(memberId: Long, method: String, uri: String) :
    AbstractHttpException("회원(ID: $memberId)의 요청이 중복되었습니다. ($method $uri)", HttpStatus.TOO_MANY_REQUESTS)